bluemira/base/reactor.py

```diff
@@ -461,23 +463,28 @@ def __init__(self, name: str, n_sectors: int):
         self.n_sectors = n_sectors
         self.start_time = time.perf_counter()
 
-    def component(
+    def _init_construction_params(
         self,
-        with_components: list[ComponentManager] | None = None,
-    ) -> Component:
-        """Return the component tree.
+        c_params: ConstructionParams | None,
+    ) -> ConstructionParams:
+        c_params = c_params or {}
+        c_params["total_sectors"] = self.n_sectors
+        c_params["component_filter"] = (
+            c_params["component_filter"]
+            if "component_filter" in c_params
+            else FilterMaterial()
+        )
+        return c_params
```

Comment:
I think I'd like this to output some immutable object, maybe a frozen dataclass?